### DISCUSSION TITLE
panic booting yocto-generated wic, prebuilt works fine

### DISCUSSION THREAD

**Discussion by mcmordie** [polarfire-soc/polarfire-soc-documentation#35] [Linux build systems for PolarFire SoC] [answered]
Building from WSL2 Ubuntu 20.04LTS host, the project (master/revision 5fba5ac95b5791d559deb95cd315372a33f80fcc) built fine from command 

`MACHINE=icicle-kit-es bitbake mpfs-dev-cli`

generating 

`tmp-glibc/deploy/images/icicle-kit-es/mpfs-dev-cli-icicle-kit-es.wic.gz`

using USBImager I rebooted and entered usbcmsc and wrote the image to eMMC.

on bootup I get the following:

```
[    4.833036] VFS: Mounted root (ext4 filesystem) readonly on device 179:3.
[    4.840584] devtmpfs: mounted
[    4.854106] Freeing unused kernel image (initmem) memory: 2132K
[    4.860293] Run /sbin/init as init process
[    4.898509] EXT4-fs warning (device mmcblk0p3): ext4_dirblock_csum_verify:376: inode #10943: comm init: No space for directory leaf checksum. Please run e2fsck -D.
[    4.913416] EXT4-fs error (device mmcblk0p3): ext4_dx_find_entry:1725: inode #10943: block 8: comm init: Directory block failed checksum
[    4.926204] EXT4-fs warning (device mmcblk0p3): ext4_dirblock_csum_verify:376: inode #10943: comm init: No space for directory leaf checksum. Please run e2fsck -D.
[    4.941085] EXT4-fs error (device mmcblk0p3): ext4_dx_find_entry:1725: inode #10943: block 8: comm init: Directory block failed checksum
[    4.953820] EXT4-fs warning (device mmcblk0p3): ext4_dirblock_csum_verify:376: inode #10943: comm init: No space for directory leaf checksum. Please run e2fsck -D.
[    4.968714] EXT4-fs error (device mmcblk0p3): ext4_dx_find_entry:1725: inode #10943: block 8: comm init: Directory block failed checksum
[    4.981431] EXT4-fs warning (device mmcblk0p3): ext4_dirblock_csum_verify:376: inode #10943: comm init: No space for directory leaf checksum. Please run e2fsck -D.
[    4.996261] EXT4-fs error (device mmcblk0p3): ext4_dx_find_entry:1725: inode #10943: block 8: comm init: Directory block failed checksum
/sbin/init: erro[    5.009237] Kernel panic - not syncing: Attempted to kill init! exitcode=0x00007f00
r while loading [    5.018304] CPU: 2 PID: 1 Comm: init Not tainted 5.15.32-linux4microchip+fpga-2022.05 #1
shared libraries[    5.027763] Hardware name: Microchip PolarFire-SoC Icicle Kit (DT)
: libseccomp.so.[    5.035308] Call Trace:
2: cannot open s[    5.039129] [<ffffffff80004820>] dump_backtrace+0x1c/0x24
hared object fil[    5.045913] [<ffffffff804c2c7c>] show_stack+0x2c/0x38
e: Error 74
[    5.052344] [<ffffffff804c597e>] dump_stack_lvl+0x40/0x58
[    5.058863] [<ffffffff804c59aa>] dump_stack+0x14/0x1c
[    5.063966] [<ffffffff804c2df6>] panic+0xe8/0x292
[    5.068712] [<ffffffff8000d8fa>] do_exit+0x6fc/0x74a
[    5.073731] [<ffffffff8000e57a>] do_group_exit+0x2a/0x7e
[    5.079095] [<ffffffff8000e5e6>] __wake_up_parent+0x0/0x20
[    5.084642] [<ffffffff80002ffe>] ret_from_syscall+0x0/0x2
[    5.090095] SMP: stopping secondary CPUs
[    5.094077] ---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x00007f00 ]---
```

What am I doing wrong?  The prebuilt image worked fine.

Dave

**Comment by vfalanis** [comment, marked as answer]
Hi Dave,

We've done a build from GitHub master branch using a couple of Linux host machines with Ubuntu 20.04 and we were not able to reproduce the issue described.

Since WSL is not officially supported by the Yocto project, that's something that we don't validate as part of our releases. It could be that there is something in WSL or LD_LIBRARY_PATH causing the issue reported.

Could you please try with a Linux host machine or a Virtual Machine running Ubuntu instead and see if this works for you?

Thanks

**Reply by mcmordie** [reply]
In principle yes I could set up a machine with Ubuntu 20.04, although given that the build completes without issue I was thinking platform is unlikely to be the cause.  core-image-minimal built this way runs just fine-- the problem seems to be with the mpfs-dev-cli target.

**Reply by vfalanis** [reply]
Hi Dave,

After looking once again at the log provided above, the first messages shown in the console indicate that the image didn't flash properly to the eMMC.

Could you please try uncompressing/extracting the mpfs-dev-cli-icicle-kit-es.wic.gz file manually using a file archiver such as 7zip and then flash the .wic file using USBimager?

**Reply by mcmordie** [reply]
Hi Valentina,

I actually went ahead and re-built everything on bona fide Ubuntu 20.04 LTS.  It turns out the issue was what you mentioned in your second reply: I was writing the .gz file with USBImager following these [instructions](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/boards/mpfs-icicle-kit-es/updating-icicle-kit/updating-icicle-kit-design-and-linux.md):

![image](https://user-images.githubusercontent.com/5338911/172480961-c9fd585b-2cf4-46c7-856b-84ebe5a86f81.png)

This does not work for me with USBImager 1.08, but extracting the .wic and writing that works fine, both with Ubuntu 20.04 LTS on bare metal and on WSL2.

Thanks for your help!

Dave
